tools/cldr-code/src/main/java/org/unicode/cldr/util/SimpleXMLSource.java

```diff
@@ -184,7 +184,7 @@ public static String normalize2(String valueToMatch, Normalizer2 normalizer2) {
             String result = replace(NON_ALPHANUM, normalizer2.normalize(temp), "");
             return result.replace('\uFFFF','\u202F');
         }
-        return replace(NON_ALPHANUM, normalizer2.normalize(valueToMatch.replace('\u202F', '\u00A0')), "");
+        return replace(NON_ALPHANUM, normalizer2.normalize(valueToMatch), "");
```

Comment:
How did this have an effect? Is `java.lang.String.replace(char)` not  non-BMP safe?

I guess it wasn't 𒀯 (U+1202F).

Reply:
BMP characters never overlap with high or low surrogates; so not a problem.